provider/core/src/export/payload.rs

```diff
@@ -17,13 +18,20 @@ trait ExportableDataPayload {
         &self,
         serializer: &mut dyn erased_serde::Serializer,
     ) -> Result<(), DataError>;
+    fn maybe_as_varule_bytes(&self) -> Option<Box<[u8]>>;
     fn as_any(&self) -> &dyn Any;
     fn eq_dyn(&self, other: &dyn ExportableDataPayload) -> bool;
 }
 
 impl<M: DynamicDataMarker> ExportableDataPayload for DataPayload<M>
 where
-    for<'a> <M::DataStruct as Yokeable<'a>>::Output: Bake + BakeSize + serde::Serialize + PartialEq,
+    for<'a> <M::DataStruct as Yokeable<'a>>::Output:
+        Bake + BakeSize + serde::Serialize + MaybeExportAsVarULE<'a> + PartialEq,
+    for<'a> <<M::DataStruct as Yokeable<'a>>::Output as MaybeExportAsVarULE<'a>>::VarULE: VarULE,
```

Comment:
I'm really not a fan of naming associated types by their bound. `...::VarULE: VarULE` does not really tell me what any of this *is*. Can you come up with clear names for these associated type, like we renamed `Yokeable` to `DataStruct`?

Reply:
It tells you what it literally _is_ (it is an associated VarULE type, which is 100% correct), just not what it _does_.

I don't have any other name that is as precise and correct as this. But if you want to brainstorm, and if you like the `DataStruct` themed naming:

- `DataStructAsVarULE`
- `OptionalVarULE`
- `CompactDataStruct`
- `DataStruct` (it is on a different trait so using the same name is possible)

Reply:
DataStructAsVarULE seems nice. but I don't think this part is overly important to get right

Reply:
Other suggestions:

- `EncodedStruct`
- `VarULEEncoded`

Agreement: `EncodedStruct`